src/dbup-cassandra/CassandraTableJournal.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using DbUp.Engine;
+using DbUp.Engine.Output;
+using DbUp.Engine.Transactions;
+using DbUp.Support;
+
+namespace dbup_cassandra
+{
+    public class CassandraTableJournal : TableJournal
+    {
+        /// <inheritdoc/>
+        public CassandraTableJournal(Func<IConnectionManager> connectionManager, Func<IUpgradeLog> logger, string schema, string table) 
+            : base(connectionManager, logger, new CassandraObjectParser(), schema?.ToLower(), table)
+        {
+        }
+
+        protected override string GetInsertJournalEntrySql(string @scriptName, string @applied)
+        {
+            return $"insert into {FqSchemaTableName} (id, script_name, applied) values ({Guid.NewGuid()},:" + scriptName.Replace("@", "") + ",:" + applied.Replace("@", "") + ")";
+        }
+
+        protected override string GetJournalEntriesSql()
+        {
+            return $"select script_name from {FqSchemaTableName}";
+        }
+
+        protected override string CreateSchemaTableSql(string quotedPrimaryKeyName)
+        {
+            return
+$@"create table {FqSchemaTableName} (
+    id UUID PRIMARY KEY,
+    script_name text,
```

Comment:
Does this need to be a partition key?

Reply:
My thinking was id being the sole partition key. Due to value being generated the partitions should rotate around. Given each script name will be unique again partitions would rotate around. What is the benefit of the composite partition key?

Let's chat.